src/releases_service.ts

```diff
@@ -0,0 +1,94 @@
+import {AppInfo, DownloadInfo} from './types'
+import {ActionsCore} from './adapters/core'
+import {Environment} from './adapters/environment'
+import {
+  Octokit,
+  ReposListReleasesItem,
+  ReposListReleasesResponseData
+} from './adapters/octokit'
+import * as semver from 'semver'
+
+export class ReleasesService {
+  private _env: Environment
+  private _core: ActionsCore
+  private _octokit: Octokit
+
+  constructor(env: Environment, core: ActionsCore, octokit: Octokit) {
+    this._env = env
+    this._core = core
+    this._octokit = octokit
+  }
+
+  async getDownloadInfo(app: AppInfo): Promise<DownloadInfo> {
+    const assetName = this.getAssetName(app.name)
+    const repo = {owner: 'k14s', repo: app.name}
+
+    const response = await this._octokit.repos.listReleases(repo)
+    const releases: ReposListReleasesResponseData = response.data
+
+    if (app.version == 'latest') {
+      const release = this.sortReleases(releases)[0]
+      this._core.debug(`Using latest version for ${app.name} (${release.name})`)
+      return this.getDownloadInfoForAsset(app, assetName, release)
+    }
+
+    for (const candidate of releases) {
+      if (candidate.name == app.version) {
+        return this.getDownloadInfoForAsset(app, assetName, candidate)
+      }
+    }
+
+    throw new Error(`Could not find version "${app.version}" for ${app.name}`)
+  }
+
+  private getDownloadInfoForAsset(
+    app: AppInfo,
+    assetName: string,
+    release: ReposListReleasesItem
+  ): DownloadInfo {
+    for (const candidate of release.assets) {
+      if (candidate.name == assetName) {
+        this._core.debug(`Found executable ${assetName} for ${describe(app)}`)
+        return {
+          version: release.name,
+          assetName: assetName,
+          url: candidate.browser_download_url,
+          releaseNotes: release.body
+        }
+      }
+    }
+    throw new Error(
+      `Could not find executable ${assetName} for ${describe(app)}`
+    )
+  }
+
+  private sortReleases(
```

Comment:
This is how we get the latest version: sort according to semver ordering, and take the highest as the latest release (to avoid treating patches for old versions as more recent).

I'm assuming any release with a valid semver name should be treated as a candidate, and any that don't match can be put to the back of the queue.